File: backend/app/schemas/backtest.py
from pydantic import BaseModel


class SymbolInfo(BaseModel):
    symbol: str
    interval: str
    days: int
    file: str


class SymbolListResponse(BaseModel):
    symbols: list[SymbolInfo]
    total: int


class BacktestRequest(BaseModel):
    symbol: str
    interval: str
    days: int
    strategy: str
    start: str | None = None
    end: str | None = None


class EquityPoint(BaseModel):
    step: int
    timestamp: int
    equity: float


class Trade(BaseModel):
    type: str
    step: int
    timestamp: int
    price: float
    pnl: float | None = None


class Metrics(BaseModel):
    total_return: float
    annualized_return: float
    annualized_vol: float
    sharpe_ratio: float
    max_drawdown: float
    win_rate: float


class BacktestMeta(BaseModel):
    strategy: str
    symbol: str
    interval: str
    bars: int


class BacktestResponse(BaseModel):
    equity_curve: list[EquityPoint]
    trades: list[Trade]
    metrics: Metrics
    meta: BacktestMeta

File: frontend/src/api/backtest.ts
// 与 backend/app/schemas/backtest.py 对齐
import client from './client'

export interface SymbolInfo {
  symbol: string
  interval: string
  days: number
  file: string
}

export interface SymbolListResponse {
  symbols: SymbolInfo[]
  total: number
}

export interface BacktestRequest {
  symbol: string
  interval: string
  days: number
  strategy: string
  start?: string
  end?: string
}

export interface EquityPoint {
  step: number
  timestamp: number
  equity: number
}

export interface Trade {
  type: string
  step: number
  timestamp: number
  price: number
  pnl: number | null
}

export interface Metrics {
  total_return: number
  annualized_return: number
  annualized_vol: number
  sharpe_ratio: number
  max_drawdown: number
  win_rate: number
}

export interface BacktestMeta {
  strategy: string
  symbol: string
  interval: string
  bars: number
}

export interface BacktestResponse {
  equity_curve: EquityPoint[]
  trades: Trade[]
  metrics: Metrics
  meta: BacktestMeta
}

export async function fetchSymbolList(): Promise<SymbolListResponse> {
  return client.get<unknown, SymbolListResponse>('/backtest/symbols')
}

export async function runBacktest(req: BacktestRequest): Promise<BacktestResponse> {
  return client.post<unknown, BacktestResponse>('/backtest/run', req)
}
